src/ADAL.PCL/Http/AdalHttpClient.cs

```diff
@@ -105,12 +111,29 @@ public AdalHttpClient(string uri, CallState callState)
                         serviceEx = new AdalServiceException(tokenResponse.Error, tokenResponse.ErrorDescription,
                             errorCodes, ex);
 
+                        if ((int)ex.WebResponse.StatusCode == 400 && tokenResponse.Error == interactionRequired)
```

Comment:
There's no need to cast to _int_ - you could just compare to the defined enum value _HttpStatusCode.BadRequest_.